app/src/main/java/de/danoeh/antennapod/dialog/EpisodesApplyActionFragment.java

```diff
@@ -407,13 +408,13 @@ private void refreshTitles() {
         mAdapter.notifyDataSetChanged();
     }
 
-    private void refreshCheckboxes() {
+    private void refreshCheckboxes(Activity activity) {
         for (int i = 0; i < episodes.size(); i++) {
             FeedItem episode = episodes.get(i);
             boolean checked = checkedIds.contains(episode.getId());
             mListView.setItemChecked(i, checked);
         }
-        ActivityCompat.invalidateOptionsMenu(EpisodesApplyActionFragment.this.getActivity());
+        activity.invalidateOptionsMenu();
```

Comment:
What is the reason for adding the activity as a parameter? Can't it just call `getActivity()` here? The method is located directly in the fragment, just like the calling ones.

Reply:
It opens the possibility for a NPE (as getActivity() is nullable). Calling getActivity() to all the `refreshCheckboxes()` methods used by the other stuff makes it so it's getting refreshCheckboxes itself with the activity as a parameter, so there's no risk of a NPE.

I'm terrible at explaining things, hopefully that made sense. I hope I explained that right.

Reply:
I don't understand. If `getActivity()` is `null`, your method also throws a NPE because it simply passes `null` to the `refreshCheckboxes` method.

Reply:
I see now, adding @NonNull to Activity activity lets all the `getActivity()` calls possibly return null. My assumption was wrong. So I think this means it's safer to just have a possible NPE on the `refreshCheckboxes` method instead of a possible NPE for every time `refreshCheckboxes()` is called. I'll use your suggestion.